pkg/injector/webhook.go

```diff
@@ -120,77 +120,75 @@ func (wh *webhook) run(stop <-chan struct{}) {
-		errmsg := fmt.Sprintf("Invalid Content-Type: %q", contentType)
-		http.Error(w, errmsg, http.StatusUnsupportedMediaType)
-		log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusUnsupportedMediaType)
+	if contentType := req.Header.Get("Content-Type"); contentType != contentTypeJSON {
+		err := errors.Errorf("Invalid content type: %s; Expected %s", contentType, contentTypeJSON)
+		http.Error(w, err.Error(), http.StatusUnsupportedMediaType)
+		log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusUnsupportedMediaType)
 		return
 	}
 
-	var body []byte
+	var admissionRequestBody []byte
 	if req.Body != nil {
 		var err error
-		if body, err = ioutil.ReadAll(req.Body); err != nil {
-			errmsg := fmt.Sprintf("Error reading request body: %s", err)
-			http.Error(w, errmsg, http.StatusInternalServerError)
-			log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusInternalServerError)
+		if admissionRequestBody, err = ioutil.ReadAll(req.Body); err != nil {
+			err := errors.Errorf("Error reading request admissionRequestBody: %s", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusInternalServerError)
 			return
 		}
 	}
 
-	if len(body) == 0 {
-		errmsg := "Empty request body"
-		http.Error(w, errmsg, http.StatusBadRequest)
-		log.Error().Msgf("Request error: error=%s, code=%v", errmsg, http.StatusBadRequest)
+	if len(admissionRequestBody) == 0 {
+		err := errors.New("Empty request admissionRequestBody")
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		log.Error().Err(err).Msgf("Responded to admission request with HTTP %v", http.StatusBadRequest)
 		return
 	}
 
 	var admissionReq v1beta1.AdmissionReview
 	var admissionResp v1beta1.AdmissionReview
-	if _, _, err := deserializer.Decode(body, nil, &admissionReq); err != nil {
-		log.Error().Err(err).Msg("Error decoding admission request")
-		admissionResp.Response = toAdmissionError(err)
+	if _, _, err := deserializer.Decode(admissionRequestBody, nil, &admissionReq); err != nil {
+		log.Error().Err(err).Msgf("Error decoding admission request body: %s", string(admissionRequestBody))
+		admissionResp.Response = admissionError(err)
 	} else {
 		admissionResp.Response = wh.mutate(admissionReq.Request)
 	}
 
 	resp, err := json.Marshal(&admissionResp)
 	if err != nil {
-		errmsg := fmt.Sprintf("Error marshalling admission response: %s", err)
-		http.Error(w, errmsg, http.StatusInternalServerError)
-		log.Error().Msgf("Request error, error=%s, code=%v", errmsg, http.StatusInternalServerError)
+		http.Error(w, fmt.Sprintf("Error marshalling admission response: %s", err), http.StatusInternalServerError)
+		log.Error().Err(err).Msgf("Responded to admission request for pod %s/%s with HTTP %v", admissionReq.Request.Namespace, admissionReq.Request.Name, http.StatusInternalServerError)
```

Comment:
how do we know this request is always pod?

Reply:
We configure the MutatingWebhookConfiguration object to only observe `create` events on `pod`;
https://github.com/openservicemesh/osm/blob/release-v0.4/pkg/injector/webhook.go#L332-L336
I'll add a comment explaining this assumption.
I also see that we do not have create/pod in the template - will add it as part of this PR for clarity.

Reply:
Improving this for clarity with the following PR:  https://github.com/openservicemesh/osm/pull/1904